excel_upload_be/Controllers/UploadZipFileController.cs

```diff
@@ -64,56 +66,51 @@ public async Task<IActionResult> Upload()
                         {
                             entry.ExtractToFile(entryPath, true);
                         }
-                        }
+                    }
                 }
 
                 Console.WriteLine("Extraction complete.");
                 string copiedFolderPath = Path.ChangeExtension(filePath,null) + "\\";
                 Console.WriteLine($"Copied .zip folder path: {copiedFolderPath}");
-                
-                folderTree = createFolderTree(copiedFolderPath);
-                jsonFolderTree = JsonSerializer.Serialize(folderTree);
-                Console.WriteLine(folderTree.Name);
-                Console.WriteLine(folderTree.Subfolders[0].Name);
-                static FolderNode createFolderTree(string folderPath)
+                folderTree = _folderTreeService.createFolderTree(copiedFolderPath);
+                var jsonFolderTree = new StringContent(JsonSerializer.Serialize(folderTree), Encoding.UTF8, "appliation/json");
+                Console.WriteLine("folder tree name: " + folderTree.Name);
+                Console.WriteLine("device name: " + folderTree.Subfolders[0].Name);
+
+                List<FolderNode> devFolders = folderTree.Subfolders;
```

Comment:
Needed this line and similar lines below to work with sub folders of the folderTree node.